src/crab/type_domain.cpp

```
@@ -17,14 +17,31 @@
 
 namespace crab {
 
-static void operator++(type_encoding_t& t) { t = static_cast<type_encoding_t>(1 + static_cast<int>(t)); }
+template <is_enum T>
+static void operator++(T& t) {
+    t = static_cast<T>(1 + static_cast<std::underlying_type_t<T>>(t));
+}
+
+std::vector<data_kind_t> iterate_kinds(const data_kind_t lb, const data_kind_t ub) {
```

cppcheck

[style] The function 'iterate_kinds' is never used.

(unusedFunction)